API-GUIDELINE.md

```diff
@@ -0,0 +1,254 @@
+         {
+           "id": "b925a817-d5ad-48bb-a407-29ad8e0649b5",
+           "productId": "9f10dcfb-5cc9-4a18-843a-c07f7e22d01f",
+           "name": {
+             "en": "SAPPHIRE"
+           },
+           "productType": {
+             "typeId": "product-type",
+             "id": "2543e1d8-4915-4f72-a3c9-1df9b1b0082d",
+             "version": 8
+           },
+           "productSlug": {
+             "en": "sapphire1421832124423"
+           },
+           "variant": {
+             "id": 1,
+             "sku": "sku_SAPPHIRE_variant1_1421832124423",
+             "prices": [
+               {
+                 "value": {
+                   "type": "centPrecision",
+                   "fractionDigits": 2,
+                   "currencyCode": "EUR",
+                   "centAmount": 2800
+                 },
+                 "id": "8da659ef-9e54-447d-9c36-84912db1848f"
+               }
+             ],
+             "images": [
+               {
+                 "url": "https://www.commercetools.com/cli/data/252542005_1.jpg",
+                 "dimensions": {
+                   "w": 1400,
+                   "h": 1400
+                 }
+               }
+             ],
+             "attributes": [],
+             "assets": []
+           },
+           "price": {
+             "value": {
+               "type": "centPrecision",
+               "fractionDigits": 2,
+               "currencyCode": "EUR",
+               "centAmount": 2800
+             },
+             "id": "8da659ef-9e54-447d-9c36-84912db1848f"
+           },
+           "quantity": 2,
+           "discountedPricePerQuantity": [],
+           "state": [
+             {
+               "quantity": 2,
+               "state": {
+                 "typeId": "state",
+                 "id": "7c2e2694-aefe-43d7-888e-6a99514caaca"
+               }
+             }
+           ],
+           "priceMode": "Platform",
+           "lineItemMode": "Standard",
+           "totalPrice": {
+             "type": "centPrecision",
+             "fractionDigits": 2,
+             "currencyCode": "EUR",
+             "centAmount": 5600
+           }
+         }
+       ],
+       "store": {
+         "typeId": "store",
+         "key": "test-key"
+       },
+       "cartState": "Active",
+       "totalPrice": {
+         "type": "centPrecision",
+         "fractionDigits": 2,
+         "currencyCode": "EUR",
+         "centAmount": 5600
+       },
+       "customLineItems": [],
+       "discountCodes": [],
+       "inventoryMode": "None",
+       "taxMode": "Platform",
+       "taxRoundingMode": "HalfEven",
+       "taxCalculationMode": "LineItemLevel",
+       "refusedGifts": [],
+       "origin": "Customer"
+     }
+
+## Naming convention
+
+### General Guidelines
+To use consistent names these are the rules to follow:
+
+ DO'S
+ - use nouns instead of verbs to expose resources
+ - use lowercase and separated by hyphens (-) for path segments like /shopping-lists
```

Comment:
looks like we use lower camel case for property names composed of more than one word?

Reply:
In path segments not. For query parameters it's something different

Reply:
sure, I mean, we need an additional bullet point about field names